### ISSUE TITLE
[COMPILER REQUEST]: 

### ISSUE THREAD

**Issue by Eldadc** [compiler-explorer/compiler-explorer#3148]
### Compiler name

IAR

### Compiler version

9.20

### Compiler language

C++

### Compiler homepage

IAR in google

### Motivation

Support embedded .

**Comment by partouf** [comment]
Can you further clarify which exact compiler you have in mind here. I am aware of IAR compilers through EWAVR, but we only have support for local installation of that on Windows, not for godbolt.org

Can you provide a link to which compiler you would like to see?

**Comment by Eldadc** [comment]
Thanks for the quick response.
The compiler : ewarm
https://www.iar.com/ewarm
Of course compiler requires license.

Question : Is there support for offline network without internet.
Ia there a docker image ?

Thanks

**Comment by partouf** [comment]
Right, we have a document to get that to work as well https://github.com/compiler-explorer/compiler-explorer/blob/main/docs/EWARM.md - try to follow that to get it to work for a local installation.

I don't think there's a Windows Docker image for Compiler Explorer, so you would have to setup one yourself.

**Comment by Eldadc** [comment]
Thanks on the info.
Regarding docker   I meant of course linux docker.
Do you have a formal version ?

**Comment by mattgodbolt** [comment]
We'd love to host IAR, but without help from the vendor it's not something we can set up ourselves — IAR requires commercial licensing and isn't freely redistributable.

If anyone has contacts at IAR who might be interested in collaborating, please reach out to @mattgodbolt!

*(Triaged with LLM assistance)*
